application/freshdesk/v0/get_all.go

```diff
@@ -0,0 +1,110 @@
+package freshdesk
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/instill-ai/component/base"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func (c *FreshdeskClient) GetAll(objectType string, pagination bool, paginationPath string) ([]TaskGetAllResponse, string, error) {
+
+	resp := []TaskGetAllResponse{}
+
+	httpReq := c.httpclient.R().SetResult(&resp)
+
+	var rawResp *resty.Response
+	var err error
+	if !pagination {
+
+		var path string
+		if objectType != "Skills" {
+			path = strings.ToLower(objectType)
+		} else {
+			path = "admin/skills"
+		}
+
+		rawResp, err = httpReq.Get(fmt.Sprintf("/%s", path))
+
+	} else {
+		rawResp, err = httpReq.Get(paginationPath)
+	}
+
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Will exist if there is a next page
+	linkHeader := rawResp.Header().Get("Link")
+
+	var nextPage string
+	if linkHeader != "" {
+		startIndex := strings.Index(linkHeader, "<")
+		endIndex := strings.Index(linkHeader, ">")
+		nextPage = linkHeader[startIndex+1 : endIndex]
+
+		return resp, nextPage, nil
+	}
+
+	return resp, "", nil
+}
+
+// Task 1: Get All
+type TaskGetAllInput struct {
+	ObjectType string `json:"object-type"`
+}
+
+type TaskGetAllResponse struct {
+	ID int64 `json:"id"`
+}
+
+type TaskGetAllOutput struct {
+	IDs      []int64 `json:"ids"`
+	IDLength int     `json:"id-length"`
+}
+
+func (e *execution) TaskGetAll(in *structpb.Struct) (*structpb.Struct, error) {
+	inputStruct := TaskGetAllInput{}
+	err := base.ConvertFromStructpb(in, &inputStruct)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert input to struct: %v", err)
+	}
+
+	resp, paginationPath, err := e.client.GetAll(inputStruct.ObjectType, false, "")
+
+	if err != nil {
+		return nil, err
+	}
+
+	outputStruct := TaskGetAllOutput{}
+	outputStruct.IDs = make([]int64, len(resp))
+	for index, value := range resp {
+		outputStruct.IDs[index] = value.ID
+	}
+
+	for paginationPath != "" {
```

Comment:
I think we should set some limitation to avoid to fetch all data if the data count is big.
And, we probably can set it to fetch the latest 100 piece of data.
And, we need to write it down in the document

Reply:
For the pagination, I decided to create another input field called "length" where the user can specify how many IDs they want to retrieve. I set a limit to this, to 500. So if the user input more than 500, an error will pop up. Let me know if you want me to reduce the number. I tried to retrieve 82 tickets ID, and there is hardly any waiting (when tested in localhost), so I thought it would be okay to set it to 500.